### PR TITLE
[otbn] Fix mismatch caused by uninitialised WDRs

### PR DESCRIPTION
**Pull request by ctopal**
For more detail regarding the problem this PR fixes: https://github.com/lowRISC/opentitan/issues/17154

### CODE REVIEW

File: hw/ip/otbn/doc/_index.md [hunk 1]
```diff
@@ -1063,8 +1063,11 @@ The following state is wiped:
 * The modulus (accessible through the MOD0 to MOD7 CSRs and the MOD WSR)
 
 The wiping procedure is a two-step process:
-* Overwrite the state with randomness from URND.
-* Overwrite the state with zeros.
+* Overwrite the state with randomness from URND and request a reseed of URND.
+* Overwrite the state with randomness from reseeded URND.
+
+Note that after internal secure wipe, the state of registers are undefined.
```
Comment (andreaskurth) [inline hw/ip/otbn/doc/_index.md:1069]:
```suggestion
Note that after internal secure wipe, the state of registers is undefined.
```

**Review by andreaskurth — APPROVED** [review]
LGTM, thanks @ctopal

File: sw/otbn/code-snippets/solinas384.s [hunk 2]
```diff
@@ -207,9 +212,11 @@ solinas384:
 .section .text.start
 .globl wrap_solinas384
 wrap_solinas384:
-  /* Initialize all-zero register. */
+  /* Initialise WDRs before using them */
   bn.xor w31, w31, w31
-
+  bn.xor w14, w14, w14
+  bn.xor w15, w15, w15
+  bn.xor w16, w16, w16
```
Comment (jadephilipoom) [inline sw/otbn/code-snippets/solinas384.s:219]: I'd suggest leaving the all-zero register and the initialization code separate, like:
```
/* Initialize all-zero register. */
bn.xor  w31, w31, w31

/* Initialize WDRs before using them. */
bn.xor  w14, w14, w14
bn.xor  w15, w15, w15
bn.xor  w16, w16, w16
```
Conceptually, they're a bit different, and the `w31` register is consistently initialized to zero all over OTBN code.

File: sw/otbn/code-snippets/barrett384.s [hunk 3]
```diff
@@ -227,7 +227,11 @@ barrett384:
 .section .text.start
 .globl wrap_barrett384
 wrap_barrett384:
+  /* Initialise WDRs before using them */
   bn.xor w31, w31, w31
+  bn.xor w16, w16, w16
+  bn.xor w17, w17, w17
+  bn.xor w18, w18, w18
```
Comment (jadephilipoom) [inline sw/otbn/code-snippets/barrett384.s:234]: Same comment here about `w31`.

**Review by jadephilipoom — APPROVED** [review]
Thanks for this cleanup! The behavior changed a while back but I guess we didn't realize the code snippets were relying on the registers to be initially zero.

**Comment by ctopal** [comment]
Thanks for the quick feedback @jadephilipoom and @andreaskurth. I've added both of your suggestions now

**Pull request merged**